pandas/_libs/tslibs/period.pyx

```diff
@@ -2394,6 +2395,14 @@ class Period(_Period):
                 value = str(value)
             value = value.upper()
             dt, reso = parse_time_string(value, freq)
+            try:
+                ts = Timestamp(value)
```

Comment:
Aren't we effectively doing the parsing twice, this way? (once in `parse_time_string`, once in `Timestamp`) Any way to avoid this?

Reply:
Yes - @jreback suggested integrating this a bit more into `parse_time_string`

I opened #38312 to track resolution (I'll have a go at it if no one else does)

Reply:
Thanks!